### PR TITLE
Improve performance in JSON reader when `mixed_types_as_string` option is enabled

### PR DESCRIPTION
**Pull request by shrshi**
## Description
Addresses #15196 by applying a patch from @karthikeyann to skip the `infer_column_type_kernel` by forcing the mixed types column to be a string. 
With this optimization, we see a significant improvement in performance. Please refer to the [comment](https://github.com/rapidsai/cudf/pull/15236#issuecomment-1979772672) for a visualization of the results before and after applying this optimization as obtained from the [JSON lines benchmarking exercise](https://github.com/rapidsai/cudf/pull/15124). 

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by shrshi** [comment]
Benchmarks were run on A100 80GB GPU, with all combinations of `normalize_single_quotes, row_selection, recovery_mode, mixed_type_as_string options` being enabled/disabled.

mixed_type
mixed_type_patch

**Comment by vuule** [comment]
The entire 30x slowdown was in the type inference?

**Comment by shrshi** [comment]
Yes, the slowdown seems to be due to long string sizes when mixed types is enabled. If we enforce the column to be a string in the JSON tree parser, we can skip the type inference kernel.

**Review by karthikeyann — APPROVED** [review]

**Review by vuule — APPROVED** [review]

File: cpp/src/io/json/json_column.cu [hunk 1]
```diff
@@ -677,6 +677,9 @@ void make_device_json_column(device_span<SymbolT const> input,
           columns.try_emplace(this_col_id, columns.at(old_col_id));
```
Comment (karthikeyann) [inline cpp/src/io/json/json_column.cu:677]:
```suggestion
          col.forced_as_string_column = true;
          columns.try_emplace(this_col_id, columns.at(old_col_id));
```

File: cpp/src/io/json/json_column.cu [hunk 2]
```diff
@@ -677,6 +677,9 @@ void make_device_json_column(device_span<SymbolT const> input,
...
           continue;
         }
+        col.forced_as_string_column = true;
+        columns.try_emplace(this_col_id, columns.at(old_col_id));
+        continue;
```
Comment (karthikeyann) [inline cpp/src/io/json/json_column.cu:682]:
```suggestion
```

**Pull request merged**